app/views/orgs/rosters/show.html.erb

```diff
@@ -12,8 +12,12 @@
             Classroom roster
           </span>
           <div class='flex-justify-between'>
-            <span class="btn btn-primary btn-sm", role="button" data-remodal-target="new-student-modal">Update students</span>
-             <span class="btn btn-primary btn-sm", role="button" data-remodal-target="download-csv-modal">Download</span>
+            <% if current_organization.lti_configuration %>
+              <%= link_to "Sync from LMS", import_from_lms_roster_path, remote: true, method: :get, class: "btn btn-primary btn-sm" %>
+            <% else %>
+              <span class="btn btn-primary btn-sm", role="button" data-remodal-target="new-student-modal">Update students</span>
```

Comment:
With this, teachers won't be able to manually add students to their roster when they have an LTI tool configured. That _could_ be a design decision we're OK with for now, but what about in the future when someone might link an LTI tool for something other than roster support? (E.g. they want to link assignments with Moodle, but don't have rosters set up there). Now they can't use the roster feature 😢 

This isn't a problem yet, and I think while roster support is the only LTI feature available, this actually makes a lot of sense. We'll need to do some investigation in the future though to figure out how to handle this case gracefully.

Reply:
Yeah, I totally agree this is something worth greater discussion! 

From a technical perspective, there currently is no difference between rosters inputted manually and imported from an LMS, so it wouldn’t cause any technical problems to even allow teachers to mix and match unrestricted (though we probably want to ensure consistency with Google Classroom). 

In the future, for assignment or submission syncing, the flows will ultimately depend upon the GitHub student whose assignment it is to be identifiable to a LMS’s `user_id`, but we don’t have store that piece of information on the roster (which we currently do for Google Classroom) and (I’d actually prefer, to avoid making the roster something different than originally designed) store that information separately on the user scoped to the linked classroom org.

In the meantime, do we want instructors to add students manually in addition to being able to sync, or do we want to leave as is?

Reply:
Oh that is a good point @d12. Hmm I think for now, we shouldn't allow teachers to manually input new students. Because that will leave us with some weird data.

Reply:
To clarify: from an LMS perspective, we currently do the act of adding students to the roster via the standard #add_students action that manual entry also uses on the backend, so there is nothing technical keeping us from allowing them to mix. For Google Classroom, we tie those roster entries to a google client id, and all (or none) of those roster entries are either linked to Google or not, and mixing would produce inconsistent data.

Reply:
I'd like to strive for consistency between the two features whenever possible. Can we disallow manually adding students when either LTI or Google is enabled?

Reply:
For LMS import specifically, do you mean:

1. If an LTI Configuration exists for the classroom, and a roster hasn't yet even been created, we only allow a teacher to create a roster via import from LMS

OR

2. If an LTI Configuration exists for the classroom, and a roster hasn't yet even been created, we allow the teacher to choose between manual entry or LMS import, and depending on what they choose means that method going forward will be the only way to add to their existing roster?

Reply:
I like number 1 the best as it makes more sense. If we allow them to manually input, then they can technically also sync from LMS. Which doesn't really make sense, since I added the check to make sure no other configuration and roster currently exists.

Reply:
:+1:, # 1 sounds good. We'll need to tackle this in the future as we add more integration features, but in the mean time I think this is the most simple way to implement.

Reply:
#1 it is, then! 

I also just want to clarify that option #2 would avoid mixing manual/imported entries. We would just be letting the user choose whether they want a manual or imported entries upon first sync, which would cover the scenario @d12 first described and also keep roster entries from intermingling, as @stephaniegiang brought up.